core/services/ocrbootstrap/delegate.go

```diff
@@ -118,14 +118,15 @@ func (d *Delegate) ServicesForSpec(jobSpec job.Job, qopts ...pg.QOpt) (services
 		if routerFields.ContractVersion != 1 || routerFields.ContractUpdateCheckFrequencySec == 0 {
 			return nil, errors.New("invalid router contract config")
 		}
-		configProvider, err = relayer.NewFunctionsProvider(
+		configProvider, err = relayer.NewPluginProvider(
 			ctx,
 			types.RelayArgs{
 				ExternalJobID: jobSpec.ExternalJobID,
 				JobID:         spec.ID,
 				ContractID:    spec.ContractID,
 				RelayConfig:   spec.RelayConfig.Bytes(),
 				New:           d.isNewlyCreatedJob,
+				ProviderType:  types.ProviderTypeFunctions,
```

Comment:
I'm still not 100% sure that I understand this new `types.ProviderType*`. Where will the mapping from plugin type to provider type exist long-term? Core cannot continue to make this associate explicit in the delegate like this. To be product agnostic, it would have to just pass along information from the job spec. It seems like we either need to add this new field to the jobspec, or infer/derive from the plugin type _inside the relayer_ instead of here in core.

Reply:
I understand that we don't want to lock in plugin type and provider type as 1-1, but we also have to balance that against node op UX simplicity, so I would favor hiding this difference for now.

Reply:
I had a similar reaction.

It seems like the job spec is the string that ties all the services together and so the provider type should be coming from there.

how does core know which services to start today for say, mercury vs functions? is something in the jobspec or is it inferred from something else?

my real question is: if we add the provider field to the spec, is it a breaking change or can we infer the correct behavior and backfill on-the-fly as needed to update the spec?

Reply:
>how does core know which services to start today for say, mercury vs functions? is something in the jobspec or is it inferred from something else?

That comes from the plugin type on the job spec at the moment.

>my real question is: if we add the provider field to the spec, is it a breaking change or can we infer the correct behavior and backfill on-the-fly as needed to update the spec?

For old job types of the OCR2 job type we could backfill on the fly. I don't know that we could do that in this particular case (ocrbootstrap) since I'm not sure if the pluginType is used. For functions I assume it isn't since they're checking the router DonID rather than the pluginType.